### PR TITLE
Add REST implementation for Validator's `GetDuties`

### PR DESCRIPTION
**Pull request by PatriceVignola**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

It adds the `GetDuties` REST API endpoint for the validator.

### CODE REVIEW

File: validator/client/beacon-api/duties.go [hunk 1]
```diff
@@ -5,21 +5,228 @@ import (
...
+
+	// Sync committees are an Altair feature
+	fetchSyncDuties := in.Epoch >= params.BeaconConfig().AltairForkEpoch
+
+	currentEpochDuties, err := c.getDutiesForEpoch(ctx, in.Epoch, multipleValidatorStatus, true, fetchSyncDuties)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get duties for current epoch `%d`", in.Epoch)
+	}
+
+	// We don't fetch proposer duties for epoch+1
```
Comment (rkapka) [inline validator/client/beacon-api/duties.go:48]: Actually we do: https://github.com/prysmaticlabs/prysm/blob/9f44d6e45292315f6103159d2ef88d9c0c900ea0/beacon-chain/rpc/eth/validator/validator.go#L153

File: validator/client/beacon-api/duties.go [hunk 2]
```diff
@@ -45,3 +252,45 @@ func (c beaconApiDutiesProvider) GetAttesterDuties(ctx context.Context, epoch ty
...
 }
+
+// GetProposerDuties retrieves the proposer duties for the given epoch
+func (c beaconApiDutiesProvider) GetProposerDuties(ctx context.Context, epoch types.Epoch) ([]*apimiddleware.ProposerDutyJson, error) {
+	proposerDuties := apimiddleware.ProposerDutiesResponseJson{}
+	if _, err := c.jsonRestHandler.GetRestJsonResponse(ctx, fmt.Sprintf("/eth/v1/validator/duties/proposer/%d", epoch), &proposerDuties); err != nil {
+		return nil, errors.Wrapf(err, "failed to query proposer duties for epoch `%d`", epoch)
+	}
+
+	for index, proposerDuty := range proposerDuties.Data {
```
Comment (rkapka) [inline validator/client/beacon-api/duties.go:263]: You may want to check if `proposerDuties.Data == nil` first, like you do in `GetCommittees`.

File: validator/client/beacon-api/duties.go [hunk 3]
```diff
@@ -45,3 +252,45 @@ func (c beaconApiDutiesProvider) GetAttesterDuties(ctx context.Context, epoch ty
...
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal validator indices")
+	}
+
+	syncDuties := apimiddleware.SyncCommitteeDutiesResponseJson{}
+	if _, err := c.jsonRestHandler.PostRestJson(ctx, fmt.Sprintf("/eth/v1/validator/duties/sync/%d", epoch), nil, bytes.NewBuffer(validatorIndicesBytes), &syncDuties); err != nil {
+		return nil, errors.Wrap(err, "failed to send POST data to REST endpoint")
+	}
+
+	for index, syncDuty := range syncDuties.Data {
```
Comment (rkapka) [inline validator/client/beacon-api/duties.go:289]: You may want to check if `syncDuties.Data == nil` first, like you do in `GetCommittees`.

**Review by rkapka — CHANGES_REQUESTED** [review]

File: validator/client/beacon-api/duties.go [hunk 4]
```diff
@@ -5,21 +5,228 @@ import (
...
+	}
+
+	return duties, nil
+}
+
+// GetCommittees retrieves the committees for the given epoch
+func (c beaconApiDutiesProvider) GetCommittees(ctx context.Context, epoch types.Epoch) ([]*apimiddleware.CommitteeJson, error) {
+	committeeParams := url.Values{}
+	committeeParams.Add("epoch", strconv.FormatUint(uint64(epoch), 10))
+	committeesRequest := buildURL("/eth/v1/beacon/states/head/committees", committeeParams)
```
Comment (james-prysm) [inline validator/client/beacon-api/duties.go:209]: I want to double check if using the head in this case is correct, the grpc one read a little differently.

Comment (PatriceVignola) [reply validator/client/beacon-api/duties.go:209]: The way I understand it from the spec, the `epoch` query parameter decides which epoch to query the committees for. The `head` committees will only be fetches if `epoch` isn't specified.

epoch | Fetch committees for the given epoch. If not present then the committees for the epoch of the state will be obtained.
-- | --

Comment (james-prysm) [reply validator/client/beacon-api/duties.go:205]: got it this makes sense.

**Review by james-prysm — APPROVED** [review]
will wait for radek for a second approval.

**Review by rkapka — APPROVED** [review]

**Pull request merged**